maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -347,84 +351,62 @@ public void addTestCompileSourceRoot( String path )
         return testCompileSourceRoots;
     }
 
+    @Deprecated
```

Comment:
It would be nice to document what to use instead (and why) via JavaDoc `@deprecated ...`.

Reply:
I'm willing to provide more doc once there's a consensus.
Honestly, I'm not even sure we need such an API, unless there's a clear path for the deprecation and for moving to a more immutable `MavenProject`.   What are the deprecation rules ? Will 4.0 remove things deprecated in 3.x ? If they stay forever, there's really no point in deprecating and trying to find slightly better APIs...  I'm just asking here, but 3.0 is very old I think there are quite a few things that are deprecated since early 3.x versions, which looks like 10 years or so.